marklogic-data-hub/src/test/java/com/marklogic/hub/impl/HubConfigImplTest.java

```diff
@@ -129,6 +131,15 @@ void hostOnAppConfigShouldBeUpdated() {
             "LoadSchemasCommand requires that the schemas path be absolute; path: " + config.getAppConfig().getSchemaPaths());
     }
 
+    @Test
+    void connectionTypeIsGatewayInProvisionedEnvironment(){
+        HubConfigImpl hubConfig = new HubConfigImpl("localhost", "admin", "admin");
+        Assertions.assertNull(hubConfig.getAppConfig().getRestConnectionType());
+        hubConfig.setIsProvisionedEnvironment(true);
+        hubConfig.applyProperties(new Properties());
+        Assertions.assertEquals(DatabaseClient.ConnectionType.GATEWAY, hubConfig.getAppConfig().getRestConnectionType());
```

Comment:
Looks like this should assert on appServicesConnectionType as well? Though is that actually required for the ml-gradle tasks to work? Those tasks will only use the appServices port if the user specifies a "-Pdatabase=". And that path doesn't always work well due to a bug in the REST API. The ml-gradle docs tell the user to override databaseClient directly as a better approach.